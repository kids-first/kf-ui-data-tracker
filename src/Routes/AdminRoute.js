import React from 'react';
import jwtDecode from 'jwt-decode';
import {Route, Redirect, withRouter} from 'react-router-dom';
import {Amplitude} from '@amplitude/react-amplitude';

export const isAdmin = () => {
  const token = localStorage.getItem('accessToken');
  if (token == null) {
    return false;
  }

  const decoded = jwtDecode(token);
  if (decoded.exp < Math.floor(new Date().getTime() / 1000)) return false;

  // Extract roles based on whether the token is from auth0
  const roles = decoded.context
    ? decoded.context.user.roles
    : decoded['https://kidsfirstdrc.org/roles'];

  return roles.includes('ADMIN');
};

/**
 * A route that only allows access to admin users, otherwise routes to the
 * index page
 */
const AdminRoute = ({component: Component, path, ...rest}) => {
  let comp = <Component />;

  return (
    <Amplitude
      eventProperties={{
        view: comp.type.WrappedComponent
          ? comp.type.WrappedComponent.name
          : comp.type.name,
        route: path,
      }}
    >
      <Route
        {...rest}
        path={path}
        render={props =>
          isAdmin() ? <Component {...props} /> : <Redirect to="/" />
        }
      />
    </Amplitude>
  );
};

export default withRouter(AdminRoute);
