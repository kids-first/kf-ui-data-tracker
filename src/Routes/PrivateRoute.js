import React from 'react';
import jwtDecode from 'jwt-decode';
import {Route, Redirect, withRouter} from 'react-router-dom';
import {Amplitude} from '@amplitude/react-amplitude';

export const hasToken = () => {
  const token = localStorage.getItem('accessToken');
  if (token == null) {
    return false;
  }
  return jwtDecode(token).exp > Math.floor(new Date().getTime() / 1000);
};

const PrivateRoute = ({component: Component, ...rest}) => {
  // note: unsure if this mounts the
  let comp = <Component />;

  return (
    <Amplitude
      eventProperties={{
        view: comp.type.WrappedComponent
          ? comp.type.WrappedComponent.name
          : comp.type.name,
        route: rest.path,
      }}
    >
      <Route
        {...rest}
        render={props =>
          hasToken() ? (
            <Component {...props} />
          ) : (
            <Redirect
              to={{
                pathname: '/login',
                state: {from: window.location.pathname},
              }}
            />
          )
        }
      />
    </Amplitude>
  );
};

export default withRouter(PrivateRoute);
