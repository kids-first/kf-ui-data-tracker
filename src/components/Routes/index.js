import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import PrivateRoute from './PrivateRoute';

import {
  LoginView,
  InvestigatorListView,
  FileUploadView,
  FileListView,
} from '../../views';

const Routes = () => (
  <Router>
    <Switch>
      <Route path="/login" component={LoginView} />
      <PrivateRoute exact path="/" component={InvestigatorListView} />
      <PrivateRoute path="/file-upload" component={FileUploadView} />
      <PrivateRoute path="/file-list" component={FileListView} />
    </Switch>
  </Router>
);

export default Routes;
