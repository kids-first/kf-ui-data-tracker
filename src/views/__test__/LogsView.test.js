import React from 'react';
import wait from 'waait';
import {render, act, fireEvent, cleanup} from 'react-testing-library';
import {MockedProvider} from '@apollo/react-testing';
import {MemoryRouter} from 'react-router-dom';
import {mocks} from '../../../__mocks__/kf-api-study-creator/mocks';
import myProfile from '../../../__mocks__/kf-api-study-creator/responses/myProfile.json';
import Routes from '../../Routes';

jest.mock('auth0-js');
afterEach(cleanup);

it('renders study logs view correctly', async () => {
  const tree = render(
    <MockedProvider
      mocks={[mocks[11], mocks[37], mocks[1], mocks[8]]}
      resolvers={{
        Query: {
          myProfile: _ => myProfile.data.myProfile,
        },
      }}
    >
      <MemoryRouter initialEntries={['/study/SD_8WX8QQ06/logs']}>
        <Routes />
      </MemoryRouter>
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();

  await wait(10);

  expect(tree.container).toMatchSnapshot();

  // Click on the event type dropdown
  act(() => {
    fireEvent.click(tree.getByText(/Event Type/i));
  });
  await wait();

  // Click on the event type "File Version Created"
  act(() => {
    fireEvent.click(tree.getByText(/File Version Created/i));
  });

  await wait(100);

  expect(tree.container).toMatchSnapshot();
});

it('renders study logs view correctly --  showing empty view for non-admin user', async () => {
  var regularUser = myProfile.data.myProfile;
  regularUser.roles = ['USER'];
  const tree = render(
    <MockedProvider
      resolvers={{
        Query: {
          myProfile: _ => regularUser,
        },
      }}
      mocks={mocks}
    >
      <MemoryRouter initialEntries={['/study/SD_8WX8QQ06/logs']}>
        <Routes />
      </MemoryRouter>
    </MockedProvider>,
  );
  await wait(2000);
  expect(tree.container).toMatchSnapshot();
});

it('renders study logs view -- shows an error', async () => {
  const tree = render(
    <MockedProvider
      mocks={[mocks[12], mocks[2], mocks[8]]}
      resolvers={{
        Query: {
          myProfile: _ => myProfile.data.myProfile,
        },
      }}
    >
      <MemoryRouter initialEntries={['/study/SD_8WX8QQ06/logs']}>
        <Routes />
      </MemoryRouter>
    </MockedProvider>,
  );
  await wait(0);

  expect(tree.container).toMatchSnapshot();
  expect(tree.queryByText('Error')).not.toBeNull();
});
