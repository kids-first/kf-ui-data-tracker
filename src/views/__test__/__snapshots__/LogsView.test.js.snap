// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders study logs view -- shows an error 1`] = `
<div>
  <div
    class="ui large top attached menu"
  >
    <div
      class="ui container"
    >
      <div
        class="item"
      >
        <img
          alt="Kids First logo"
          src="test-file-stub"
        />
      </div>
      <a
        aria-current="page"
        class="header item"
        href="/"
      >
        Data Tracker
      </a>
      <a
        class="item"
        href="/"
      >
        Studies
      </a>
      <div
        class="right menu"
      >
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          <img
            alt="bendolly"
            class="ui avatar image"
            src="https://lh3.googleusercontent.com/-7Agefge42sw/AAAAAAAAAAI/AAAAAAAAAAA/ACHi3reWt8KI4nO_arpKCb4FzokaKnLgtg/mo/photo.jpg"
          />
          bendolly
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/profile"
              role="option"
            >
              <i
                aria-hidden="true"
                class="user icon"
              />
              Profile
            </a>
            <a
              class="item"
              href="/logout"
              role="option"
            >
              <i
                aria-hidden="true"
                class="log out icon"
              />
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page"
  >
    <div
      class="ui basic segment ui container"
    >
      <div
        class="ui icon negative message"
      >
        <i
          aria-hidden="true"
          class="warning circle icon"
        />
        <div
          class="content"
        >
          <div
            class="header"
          >
            Error
          </div>
          <p>
            Network error: something went wrong
          </p>
        </div>
      </div>
    </div>
    <div
      class="ui basic segment ui container"
    >
      <div
        class="ui icon negative message"
      >
        <i
          aria-hidden="true"
          class="warning circle icon"
        />
        <div
          class="content"
        >
          <div
            class="header"
          >
            Error
          </div>
          <p>
            Event Error: 
            Network error: something went wrong with your event logs request
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui segment footer"
  >
    <div
      class="ui container text-10"
    >
      <div
        class="ui horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui basic horizontal label text-10"
          />
        </div>
        <div
          class="item noMargin"
          role="listitem"
        >
          UI
           
          <a
            href="https://github.com/kids-first/kf-ui-data-tracker/commit/undefined"
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders study logs view correctly --  showing empty view for non-admin user 1`] = `
<div>
  <div
    class="ui large top attached menu"
  >
    <div
      class="ui container"
    >
      <div
        class="item"
      >
        <img
          alt="Kids First logo"
          src="test-file-stub"
        />
      </div>
      <a
        aria-current="page"
        class="header item"
        href="/"
      >
        Data Tracker
      </a>
      <a
        class="item"
        href="/"
      >
        Studies
      </a>
      <div
        class="right menu"
      >
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          <img
            alt="bendolly"
            class="ui avatar image"
            src="https://lh3.googleusercontent.com/-7Agefge42sw/AAAAAAAAAAI/AAAAAAAAAAA/ACHi3reWt8KI4nO_arpKCb4FzokaKnLgtg/mo/photo.jpg"
          />
          bendolly
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/profile"
              role="option"
            >
              <i
                aria-hidden="true"
                class="user icon"
              />
              Profile
            </a>
            <a
              class="item"
              href="/logout"
              role="option"
            >
              <i
                aria-hidden="true"
                class="log out icon"
              />
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page"
  >
    <section
      id="study"
    >
      <div
        class="ui basic secondary segment"
      >
        <div
          class="ui container"
        >
          <h1
            class="ui header"
          >
            benchmark extensible e-business
          </h1>
          <button
            class="ui tiny basic icon button"
          >
            <i
              aria-hidden="true"
              class="copy icon"
            />
            SD_8WX8QQ06
          </button>
        </div>
      </div>
      <div
        class="ui container"
      >
        <div
          class="ui pink pointing secondary menu"
        >
          <a
            class="item"
            href="/study/SD_8WX8QQ06/basic-info/info"
          >
            Basic Info
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/dashboard"
          >
            Dashboard
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/documents"
          >
            Documents
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/cavatica"
          >
            Cavatica
          </a>
          <a
            aria-current="page"
            class="active item active"
            href="/study/SD_8WX8QQ06/logs"
          >
            Logs
          </a>
        </div>
      </div>
    </section>
    <div
      class="ui basic very padded segment ui container"
    >
      <h2
        class="ui disabled center aligned header"
      >
        <i
          aria-hidden="true"
          class="clock outline icon"
        />
        Coming Soon...
      </h2>
    </div>
  </div>
  <div
    class="ui segment footer"
  >
    <div
      class="ui container text-10"
    >
      <div
        class="ui horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui basic horizontal label text-10"
          />
        </div>
        <div
          class="item noMargin"
          role="listitem"
        >
          UI
           
          <a
            href="https://github.com/kids-first/kf-ui-data-tracker/commit/undefined"
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders study logs view correctly 1`] = `
<div>
  <div
    class="ui large top attached menu"
  >
    <div
      class="ui container"
    >
      <div
        class="item"
      >
        <img
          alt="Kids First logo"
          src="test-file-stub"
        />
      </div>
      <a
        aria-current="page"
        class="header item"
        href="/"
      >
        Data Tracker
      </a>
    </div>
  </div>
  <div
    class="page"
  >
    <section
      id="study"
    >
      <div
        class="ui basic secondary segment"
      >
        <div
          class="ui container"
        >
          <div
            class="ui placeholder"
          >
            <div
              class="header"
            >
              <div
                class="line"
              />
              <div
                class="line"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui container"
      >
        <div
          class="ui pink pointing secondary menu"
        >
          <a
            class="item"
            href="/study/SD_8WX8QQ06/basic-info/info"
          >
            Basic Info
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/dashboard"
          >
            Dashboard
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/documents"
          >
            Documents
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/cavatica"
          >
            Cavatica
          </a>
          <a
            aria-current="page"
            class="active item active"
            href="/study/SD_8WX8QQ06/logs"
          >
            Logs
          </a>
        </div>
      </div>
    </section>
    <div
      class="ui basic vertical segment ui container"
    >
      <div
        class="ui placeholder"
      >
        <div
          class="image header"
        >
          <div
            class="line"
          />
          <div
            class="line"
          />
        </div>
        <div
          class="paragraph"
        >
          <div
            class="line"
          />
          <div
            class="line"
          />
          <div
            class="line"
          />
          <div
            class="line"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui segment footer"
  >
    <div
      class="ui container text-10"
    >
      <div
        class="ui horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui basic horizontal label text-10"
          />
        </div>
        <div
          class="item noMargin"
          role="listitem"
        >
          UI
           
          <a
            href="https://github.com/kids-first/kf-ui-data-tracker/commit/undefined"
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders study logs view correctly 2`] = `
<div>
  <div
    class="ui large top attached menu"
  >
    <div
      class="ui container"
    >
      <div
        class="item"
      >
        <img
          alt="Kids First logo"
          src="test-file-stub"
        />
      </div>
      <a
        aria-current="page"
        class="header item"
        href="/"
      >
        Data Tracker
      </a>
      <a
        class="item"
        href="/"
      >
        Studies
      </a>
      <div
        class="right menu"
      >
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          Admin
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/tokens"
              role="option"
            >
              <i
                aria-hidden="true"
                class="key icon"
              />
              Developer Tokens
            </a>
            <a
              class="item"
              href="/cavatica-projects"
              role="option"
            >
              <i
                aria-hidden="true"
                class="folder open icon"
              />
              Cavatica Projects
            </a>
            <a
              class="item"
              href="/events"
              role="option"
            >
              <i
                aria-hidden="true"
                class="history icon"
              />
              Event Log
            </a>
          </div>
        </div>
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          <img
            alt="bendolly"
            class="ui avatar image"
            src="https://lh3.googleusercontent.com/-7Agefge42sw/AAAAAAAAAAI/AAAAAAAAAAA/ACHi3reWt8KI4nO_arpKCb4FzokaKnLgtg/mo/photo.jpg"
          />
          bendolly
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/profile"
              role="option"
            >
              <i
                aria-hidden="true"
                class="user icon"
              />
              Profile
            </a>
            <a
              class="item"
              href="/logout"
              role="option"
            >
              <i
                aria-hidden="true"
                class="log out icon"
              />
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page"
  >
    <section
      id="study"
    >
      <div
        class="ui basic secondary segment"
      >
        <div
          class="ui container"
        >
          <h1
            class="ui header"
          >
            benchmark extensible e-business
          </h1>
          <button
            class="ui tiny basic icon button"
          >
            <i
              aria-hidden="true"
              class="copy icon"
            />
            SD_8WX8QQ06
          </button>
        </div>
      </div>
      <div
        class="ui container"
      >
        <div
          class="ui pink pointing secondary menu"
        >
          <a
            class="item"
            href="/study/SD_8WX8QQ06/basic-info/info"
          >
            Basic Info
            <div
              class="ui blue mini label"
            >
              BETA
            </div>
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/dashboard"
          >
            Dashboard
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/documents"
          >
            Documents
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/cavatica"
          >
            Cavatica
          </a>
          <a
            aria-current="page"
            class="active item active"
            href="/study/SD_8WX8QQ06/logs"
          >
            Logs
          </a>
        </div>
      </div>
    </section>
    <div
      class="ui basic vertical segment ui container"
    >
      <div
        class="ui basic right floated segment noMargin noPadding"
      >
        <div
          aria-expanded="false"
          class="ui selection dropdown"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-live="polite"
            class="default text"
            role="alert"
          >
            Event Type
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="false"
              aria-selected="true"
              class="selected item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study File Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study File Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study File Deleted
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                File Version Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                File Version Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Deleted
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Linked
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Unlinked
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Other
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Creation Start
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Creation Error
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Creation Success
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Bucket Creation Start
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Bucket Creation Error
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Bucket Creation Success
              </span>
            </div>
          </div>
        </div>
      </div>
      <h2
        class="ui header mt-6"
      >
        Event Logs
      </h2>
      <div
        class="ui very relaxed list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.799Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Katrina Torres created version FV_CZ9APDBX of file SF_ID7TZFDM
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.792Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Lisa Steele created file SF_ID7TZFDM
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.711Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Elizabeth White created version FV_IR5YTHKJ of file SF_SX4YTPYK
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.703Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Steven Morgan created version FV_HWYZ3P9S of file SF_SX4YTPYK
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.695Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Charles Booker created version FV_Z00M0H85 of file SF_SX4YTPYK
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.687Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Justin Heath created version FV_FSTR09MV of file SF_SX4YTPYK
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.678Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Roger Swanson created file SF_SX4YTPYK
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui segment footer"
  >
    <div
      class="ui container text-10"
    >
      <div
        class="ui horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui basic horizontal label text-10"
          />
        </div>
        <div
          class="item noMargin"
          role="listitem"
        >
          UI
           
          <a
            href="https://github.com/kids-first/kf-ui-data-tracker/commit/undefined"
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders study logs view correctly 3`] = `
<div>
  <div
    class="ui large top attached menu"
  >
    <div
      class="ui container"
    >
      <div
        class="item"
      >
        <img
          alt="Kids First logo"
          src="test-file-stub"
        />
      </div>
      <a
        aria-current="page"
        class="header item"
        href="/"
      >
        Data Tracker
      </a>
      <a
        class="item"
        href="/"
      >
        Studies
      </a>
      <div
        class="right menu"
      >
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          Admin
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/tokens"
              role="option"
            >
              <i
                aria-hidden="true"
                class="key icon"
              />
              Developer Tokens
            </a>
            <a
              class="item"
              href="/cavatica-projects"
              role="option"
            >
              <i
                aria-hidden="true"
                class="folder open icon"
              />
              Cavatica Projects
            </a>
            <a
              class="item"
              href="/events"
              role="option"
            >
              <i
                aria-hidden="true"
                class="history icon"
              />
              Event Log
            </a>
          </div>
        </div>
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          <img
            alt="bendolly"
            class="ui avatar image"
            src="https://lh3.googleusercontent.com/-7Agefge42sw/AAAAAAAAAAI/AAAAAAAAAAA/ACHi3reWt8KI4nO_arpKCb4FzokaKnLgtg/mo/photo.jpg"
          />
          bendolly
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/profile"
              role="option"
            >
              <i
                aria-hidden="true"
                class="user icon"
              />
              Profile
            </a>
            <a
              class="item"
              href="/logout"
              role="option"
            >
              <i
                aria-hidden="true"
                class="log out icon"
              />
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page"
  >
    <section
      id="study"
    >
      <div
        class="ui basic secondary segment"
      >
        <div
          class="ui container"
        >
          <h1
            class="ui header"
          >
            benchmark extensible e-business
          </h1>
          <button
            class="ui tiny basic icon button"
          >
            <i
              aria-hidden="true"
              class="copy icon"
            />
            SD_8WX8QQ06
          </button>
        </div>
      </div>
      <div
        class="ui container"
      >
        <div
          class="ui pink pointing secondary menu"
        >
          <a
            class="item"
            href="/study/SD_8WX8QQ06/basic-info/info"
          >
            Basic Info
            <div
              class="ui blue mini label"
            >
              BETA
            </div>
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/dashboard"
          >
            Dashboard
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/documents"
          >
            Documents
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/cavatica"
          >
            Cavatica
          </a>
          <a
            aria-current="page"
            class="active item active"
            href="/study/SD_8WX8QQ06/logs"
          >
            Logs
          </a>
        </div>
      </div>
    </section>
    <div
      class="ui basic vertical segment ui container"
    >
      <div
        class="ui basic right floated segment noMargin noPadding"
      >
        <div
          aria-expanded="false"
          class="ui selection dropdown"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-live="polite"
            class="default text"
            role="alert"
          >
            Event Type
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="false"
              aria-selected="true"
              class="selected item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study File Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study File Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study File Deleted
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                File Version Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                File Version Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Study Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Created
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Updated
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Deleted
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Linked
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Unlinked
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Other
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Creation Start
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Creation Error
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Project Creation Success
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Bucket Creation Start
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Bucket Creation Error
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Bucket Creation Success
              </span>
            </div>
          </div>
        </div>
      </div>
      <h2
        class="ui header mt-6"
      >
        Event Logs
      </h2>
      <div
        class="ui very relaxed list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <i
            aria-hidden="true"
            class="green add icon"
          />
          <div
            class="content"
          >
            <div
              class="right floated content"
            >
              <time
                datetime="2019-10-24T15:29:53.687Z"
                title="24 Oct 2019"
              >
                6 months from now
              </time>
            </div>
            Justin Heath created version FV_FSTR09MV of file SF_SX4YTPYK
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui segment footer"
  >
    <div
      class="ui container text-10"
    >
      <div
        class="ui horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui basic horizontal label text-10"
          />
        </div>
        <div
          class="item noMargin"
          role="listitem"
        >
          UI
           
          <a
            href="https://github.com/kids-first/kf-ui-data-tracker/commit/undefined"
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
